client/packages/common/src/ui/layout/tables/components/Header/ColumnPicker.tsx

```diff
@@ -55,30 +42,15 @@ export const ColumnPicker = <T extends RecordWithId>({
     setAnchorEl(null);
   };
 
-  const toggleColumn = (column: Column<T>) => {
-    const hidden = getHiddenColumns();
-    const updatedColumns = isVisible(column)
-      ? [...hidden, String(column.key)]
-      : hidden.filter(key => key !== column.key);
-
-    setHiddenColumnsConfig({
-      ...hiddenColumnsConfig,
-      [tableKey]: updatedColumns,
-    });
-  };
-
-  useEffect(() => {
-    onChange(columns.filter(isVisible));
-  }, [columns, onChange, isVisible]);
-
   return (
     <>
       <Tooltip title={t('table.show-columns')}>
         <IconButton onClick={handleClick} aria-describedby={id}>
           <ColumnsIcon
             sx={{
-              color:
-                getHiddenColumns().length > 0 ? 'secondary.main' : undefined,
+              color: !columns.every(c => columnDisplayState[String(c.key)])
+                ? 'secondary.main'
+                : undefined,
```

Comment:
I think super minor, up to you if you want to investigate or not... but for some reason this doesn't work quite right with plugin column on initial mount:

Per video, all fields are selected, but column icon is blue. Stays blue until you uncheck a recheck the Donor column? Goes back to being blue if you refresh...

https://github.com/msupply-foundation/open-msupply/assets/55115239/1d87e946-24f8-45ba-bfff-644e0303c4f7

Reply:
> I think super minor, up to you if you want to investigate or not... but for some reason this doesn't work quite right with plugin column on initial mount:

Good catch -- I got the other two cases where the column is missing from the state object, missed this one.

Easily fixed: 15f206c27

Thanks 🙏